frontend/src/components/bucket/BucketPermission.vue

```diff
@@ -0,0 +1,156 @@
+<script setup lang="ts">
+import { ref, onMounted } from 'vue';
+import { storeToRefs } from 'pinia';
+import Button from 'primevue/button';
+import Checkbox from 'primevue/checkbox';
+import Column from 'primevue/column';
+import DataTable from 'primevue/datatable';
+import { useBucketStore } from '@/store/bucketStore';
+import { Permissions } from '@/utils/constants';
+
+const props = defineProps<{
+  bucketId: string;
+}>();
+
+const bucketStore = useBucketStore();
+const { loading, permissions } = storeToRefs(useBucketStore());
+
+onMounted(() => {
+  bucketStore.getBucketPermissions(props.bucketId);
+});
+
+const updateBucketPermission = (value: any, userId: string, permCode: string) => {
+  if (value) {
+    bucketStore.addBucketPermission(props.bucketId, userId, permCode);
+  } else {
+    bucketStore.deleteBucketPermission(props.bucketId, userId, permCode);
+  }
+};
+
+const removeBucketUser = (userId: string) => {
+  bucketStore.removeBucketUser(props.bucketId, userId);
+};
+</script>
+
+<template>
+  <Button class="mt-1 mb-4"> <font-awesome-icon icon="fa-solid fa-user-plus" class="mr-1" /> Add user </Button>
```

Comment:
not sure how this button will work. I guess it does a user search and then adds person to a new row to the permissions table.

Reply:
It's a placeholder button for now, as that functionality isn't finalized. But the button itself was included in the mockups.